Library/PagedContainer/PagedContainerViewModel.swift

```diff
@@ -0,0 +1,48 @@
+import Combine
+import Foundation
+import UIKit
+
+public class PagedContainerViewModel {
+  // Internal
+  private var subscriptions = Set<AnyCancellable>()
+
+  init() {
+    self.pageTitles = self.configureWithChildrenSubject.map { controllers in
+      controllers.map { $0.title ?? "Page " }
+    }.eraseToAnyPublisher()
+
+    self.displayChildViewControllerAtIndex = Publishers.CombineLatest(
+      self.configureWithChildrenSubject,
+      self.selectedIndex.compactMap { $0 }
+    )
+    .map { (controllers: [UIViewController], index: Int) -> (UIViewController, Int)? in
+      if index < controllers.count {
+        return (controllers[index], index)
+      } else {
+        return nil
+      }
+    }.compactMap { $0 }
+    .eraseToAnyPublisher()
+  }
+
+  // Inputs
+  func viewWillAppear() {
+    if self.selectedIndex.value == nil {
+      self.didSelectPage(atIndex: 0)
```

Comment:
This tiny bit of state (in `self.selectedIndex`, a `CurrentValueSubject` instead of a straight `PassthroughSubject`), saves a lot of lines of confusing stateless code. I'm OK with that.

Reply:
I like it!